### PR TITLE
More functions and correct return type format

### PR DESCRIPTION
**Pull request by lixingwang**
Here is the list of this PR for.
1. Add Array functions
2. Add DateTime functions
3. Add boolean function
4. Add more string functions
5. Add go module file for utils function
6. Base on the discussion we made to change the return type to an object
```
 "return": {
        "type": "string"
      }
```

### CODE REVIEW

**Comment by fm-tibco** [comment]
@mellistibco does encoding a string to base64 even make sense? I think we do that in util and strings package.  Shouldn't it just be a single base64 encode in util that encodes a binary to a string and decodes a string to a binary.  Not sure it makes sense to encode a string to a string, that is not the purpose of base64 encoding.

**Comment by mellistibco** [comment]
I think i'd put encoding stuff in utils and allow it to base64 encode more than just a string typ...

**Comment by lixingwang** [comment]
@mellistibco  The encode and decode function under utils are for string only(from code).  Why not move to string category?

**Comment by mellistibco** [comment]
Let's do the opposite and make the encode/decode in utils support more than just string type?

**Comment by lixingwang** [comment]
@mellistibco  Make sense.  Will keep it in utils and remove from strings.

**Comment by lixingwang** [comment]
@mellistibco @fm-tibco Updated PR, please have another look.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/lixingwang/contrib
```
Comment (skothari-tibco) [inline go.mod]: You'll have to change this to `project-flogo/contrib` before merging.

Comment (lixingwang) [reply go.mod]: thanks, @skothari-tibco, it should not be there,  it committed by mistake. I removed it.

**Comment by mellistibco** [comment]
@fcastill note the descriptor change.

File: function/utils/descriptor.json [hunk 2]
```diff
@@ -23,7 +26,10 @@
...
           "type": "string"
         }
-      ]
+      ],
+      "return": {
+        "type": "string"
+      }
     },
     {
       "name": "decodeString",
```
Comment (mellistibco) [inline function/utils/descriptor.json]: why not just make this decode and support any data typ?

File: function/utils/descriptor.json [hunk 3]
```diff
@@ -12,7 +12,10 @@
...
       "args": [
         { }
-      ]
+      ],
+      "return": {
+        "type": "string"
+      }
     },
     {
       "name": "encodeString",
```
Comment (mellistibco) [inline function/utils/descriptor.json]: Same comment, why not just encode and support any?

File: function/utils/descriptor.json (hunk 2 above)
Comment (lixingwang) [reply function/utils/descriptor.json]: @mellistibco How we support any type for decode output?   we received base64 data and encode to bytes and convert to what then?   I think return bytes should be good.

File: function/utils/descriptor.json (hunk 3 above)
Comment (lixingwang) [reply function/utils/descriptor.json]: It makes sense to have any type support for encode/decode's input which I have changed.

File: function/utils/descriptor.json [hunk 4]
```diff
@@ -12,29 +12,38 @@
...
         {
-          "name": "inputstring",
-          "type": "string"
+          "name": "input",
+          "type": "any"
         }
-      ]
+      ],
+      "return": {
+        "type": "bytes"
```
Comment (fm-tibco) [inline function/utils/descriptor.json]: Output of encode should be string, not 100% convince the input should be any.  Most base64 functions in most languages take a byte array

Comment (fm-tibco) [reply function/utils/descriptor.json]: @mellistibco why any?  the other problem any cause in the encode/decode are not symmetric

File: function/utils/descriptor.json [hunk 5]
```diff
@@ -12,29 +12,38 @@
...
-      "name": "decodeString",
-      "description": "decodeString returns the string represented by the base 64 encoded input string.  utils.decodeString(\"SGVsbG8sIFdvcmxk\")",
+      "name": "decodeBase64",
+      "description": "decodeBase64 returns the bytes represented by the base 64 encoded input.  utils.decodeBase64(corece.ToBytes(\"SGVsbG8sIFdvcmxk\"))",
       "varArgs": true,
       "args": [
         {
-          "name": "inputstring",
-          "type": "string"
+          "name": "input",
```
Comment (fm-tibco) [inline function/utils/descriptor.json]: the input to decode base64 should be a string.

Comment (fm-tibco) [reply function/utils/descriptor.json]: also var it should not be var args

**Pull request merged**